lib/async/task.rb

```diff
@@ -92,9 +92,9 @@ def sleep(duration = nil)
 			super
 		end
 		
-		# @deprecated Replaced by {Scheduler#timeout_after}.
-		def with_timeout(timeout, exception = TimeoutError, message = "execution expired", &block)
-			Fiber.scheduler.timeout_after(timeout, exception, message, &block)
+		# Execute the given block of code, raising the specified exception if it exceeds the given duration during a non-blocking operation.
+		def with_timeout(duration, exception = TimeoutError, message = "execution expired", &block)
+			Fiber.scheduler.with_timeout(timeout, exception, message, &block)
```

Comment:
Perhaps duration/timeout is a typo here. Failed the tests, I guess?

Reply:
Yes! Thanks :)